### DISCUSSION TITLE
Closing dangling sessions

### DISCUSSION THREAD

**Discussion by d6stringer** [FreeOpcUa/python-opcua#1372] [Q&A] [answered]
I am very new to this so forgive my ignorance. 
I'm working with a Siemens S7-1200 OPC server and I'm trying to gather some info from it. That part is going relatively smoothly. I have another question about my subscription that I will post separately/later if I don't solve it on my own.

The issue I'm having is that if I kill my client the session stays active on the server, so with a maximum session count of 5 I'm running into where my server rejects a new connection. I'm trying to come up with a way to check for and close out those dangling sessions. 

I've figured out a way to check the CurrentSessionCount:
```python
session_node = client.get_node("i=2277") # i=2277 is the CurretSessionCount
session_count = session_node.get_value()
print("Session count is: ", session_count)
```
Which is great, tells me how close I am to my limit. But that's about as much as I can do at this point. What I'd like to do is check for sessions and close any old/unused ones.

Alternatively, is there a way to rejoin a session already established? I have tried "activate_session" but that seems like it's just opening up a new one again.

Another consideration is getting old sessions to time out but I'm not sure that can be done.

thanks in advance.

**Comment by AndreasHeine** [comment, marked as answer]
the session is stored serverside the only way to get around is to shorten sessiontimeout in the s7

**Reply by d6stringer** [reply]
I've changed the timeout on the S7 and that does help. Thanks! (for whatever reason it's not letting me mark this as an answer)

Now that I've changed the timeout on the S7 though, it's saying this at the connect:
"Requested session timeout to be 3600000ms, got 360000ms instead"

I thought for sure I could fix that w/:
```python
client = Client("opc.tcp://192.168.0.1:4840", timeout=360000)
```
But that doesn't seem to change anything.

**Reply by AndreasHeine** [reply]
while connect server and client negotiate the timeout!

**Reply by d6stringer** [reply]
I found these in the Client class and altered them:
```python
self.secure_channel_timeout = 360000  # was 1 hour, changed to 5 minutes
self.session_timeout = 360000  # was 1 hour changed to 5 minutes
```
that seems to have done the trick. only question is did I do something bad?

**Reply by AndreasHeine** [reply]
no its just a warning that the server did not accept the requested client timeouts and send the client the once to use ;)

**Reply by AndreasHeine** [reply]
`self.session_timeout != response.RevisedSessionTimeout`

https://github.com/FreeOpcUa/python-opcua/blob/3012acf9047aad3acc2674e9008f0d1eee680c25/opcua/client/client.py#L404

**Reply by d6stringer** [reply]
A little derp on that last comment. you can easily edit 
```
client.session_timeout 
```
inline and it accomplishes the same thing as changing the class (duh).

**Comment by AndreasHeine** [comment]
reuse of session is not implemented!
